code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -2863,6 +2864,9 @@
 		need_mob_update += drinker.adjustFireLoss(-2 * REM * seconds_per_tick, updating_health = FALSE)
 		if(drinker.blood_volume < BLOOD_VOLUME_NORMAL)
 			drinker.blood_volume += 3 * REM * seconds_per_tick
+		if(drinker.mob_biotypes & MOB_ROBOTIC) // Bubber edit starrt
+			var/heal_amount = -2 * REM * seconds_per_tick //might as well keep this line as long as the previous
+			drinker.heal_bodypart_damage(heal_amount, heal_amount, required_bodytype = BODYTYPE_ROBOTIC) //bubber edit end
```

Comment:
add need_mob_update += to this too, maybe modularize if you'd like, a extra isheretic check would not hurt too much

Reply:
Gonna dot his today, been busy lately